### PR TITLE
UXG-1272 listings general guided setup page

### PR DESCRIPTION
**Pull request by nicmarson**
http://jira.hq.idx.org/browse/UXG-1272

### CODE REVIEW

**Review by sgreenl5 — CHANGES_REQUESTED** [review]
The save actions are inside of the `general` module's actions file, and instead should be in their corresponding actions file. The `general` module is for routes `guided-setup/connect/general` and `settings/general` only

**Review by sgreenl5 — APPROVED** [review]

File: src/vue/backend/src/mixins/guidedSetup.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    methods: {
+        goBackStep: function () {
+            this.$router.go(-1)
+        },
+        goSkipStep: function () {
+            this.$router.push({ path: this.skipPath })
+        },
+        async goContinue () {
+            await this.saveGeneralListingsSettings()
```
Comment (sgreenl5) [inline src/vue/backend/src/mixins/guidedSetup.js:20]: This is too specific of an api call an cannot be reused, for now you can take this piece and abstract it to the view level. So the view would look like
```
@continue="something"

async something () {
    await this.savegeneralsettings()
    this.goContinue()
}
```

File: src/vue/backend/src/views/guided-setup/ListingsGeneral.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<template>
+    <GuidedSetupContentCard
+        :cardTitle="cardTitle"
+        :steps="guidedSetupSteps"
+        :relatedLinks="links"
+        @back-step="goBackStep"
+        @skip-step="goSkipStep"
+        @continue="goContinue">
+        <template v-slot:controls>
+            <ListingsGeneral/>
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/ListingsGeneral.vue:10]: This does not match the new form content flow implementation, and is not adjusting the state

File: src/vue/backend/src/views/guided-setup/ListingsGeneral.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import GuidedSetupContentCard from '@/templates/GuidedSetupContentCard.vue'
+export default {
+    mixins: [GuidedSetupMixin],
+    components: {
+        ListingsGeneral,
+        GuidedSetupContentCard
+    },
+    computed: {
+        ...mapState({
+            enabled: state => state.listingsSettings.enabled,
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/ListingsGeneral.vue:28]: What are you using enabled for?

**Review by sgreenl5 — CHANGES_REQUESTED** [review]

File: src/vue/backend/src/views/guided-setup/ListingsGeneral.vue [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            defaultState: state => state.listingsSettings.defaultState
+        })
+    },
+    methods: {
+        ...mapActions({
+            setItem: 'listingsSettings/setItem',
+            progressStepperUpdate: 'progressStepper/progressStepperUpdate',
+            saveGeneralListingsSettings: 'listingsSettings/saveGeneralListingsSettings'
+        }),
+        async goContinue () {
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/ListingsGeneral.vue:51]: You already have the goContinue defined from your mixin

**Review by sgreenl5 — APPROVED** [review]

**Review by ralls — APPROVED** [review]

**Pull request merged**